### PR TITLE
Delegation03

### PR DESCRIPTION
**Pull request by mattias-p**
Implements zonemaster/zonemaster#621.
There's a discrepancy between how Engine determines parent zone and what the above PR prescribes. I suggest that this PR is accepted as is and that the discrepancy is dealt with either by changing the spec to match the implementation, or (if the wording in the implementation is strictly better) that an issue is created for Engine to add a unit test for this edge case.

### CODE REVIEW

File: lib/Zonemaster/Engine/Test/Delegation.pm [hunk 1]
```diff
@@ -353,50 +355,35 @@ sub delegation02 {
...
+        my $rr = Zonemaster::LDNS::RR->new( sprintf( q{%s IN NS %s}, $zone->name, $nsname ) );
         $p->unique_push( q{authority}, $rr );
     }
-
-    if ( @needs_v4_glue ) {
-        my $ns = $needs_v4_glue[0];
-        my $rr = Zonemaster::LDNS::RR->new( sprintf( q{%s IN A %s}, $ns->name, $ns->address->short ) );
+    if ( List::Util::all { $parent->name->is_in_bailiwick( $_->name ) } @nss_v4 ) {
+        my ( $shortest_v4 ) = sort { length( $a->name->string ) <=> length( $b->name->string ) } @nss_v4;
+        my $rr = Zonemaster::LDNS::RR->new( sprintf( q{%s IN A %s}, $shortest_v4->name, $shortest_v4->address->short ) );
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Delegation.pm]: The lenght of the name is irrelevant, isn't it?

File: lib/Zonemaster/Engine/Test/Delegation.pm [hunk 2]
```diff
@@ -353,50 +355,35 @@ sub delegation02 {
...
         $p->unique_push( q{additional}, $rr );
     }
-
-    if ( @needs_v6_glue ) {
-        my $ns = $needs_v6_glue[0];
-        my $rr = Zonemaster::LDNS::RR->new( sprintf( q{%s IN AAAA %s}, $ns->name, $ns->address->short ) );
+    if ( List::Util::all { $parent->name->is_in_bailiwick( $_->name ) } @nss_v6 ) {
+        my ( $shortest_v6 ) = sort { length( $a->name->string ) <=> length( $b->name->string ) } @nss_v6;
+        my $rr = Zonemaster::LDNS::RR->new( sprintf( q{%s IN AAAA %s}, $shortest_v6->name, $shortest_v6->address->short ) );
```
Comment (matsduf) [inline lib/Zonemaster/Engine/Test/Delegation.pm]: The lenght of the name is irrelevant, isn't it?

File: lib/Zonemaster/Engine/Test/Delegation.pm (hunk 1 above)
Comment (mattias-p) [reply lib/Zonemaster/Engine/Test/Delegation.pm]: You're right. Fixing that.
